source/features/restore-file.tsx

```diff
@@ -25,6 +23,17 @@ const getBaseReference = onetime(async (): Promise<string> => {
 	`);
 	return repository.pullRequest.baseRefOid;
 });
+const getHeadReference = async (): Promise<string> => {
```

Comment:
This is cached by the `mem` function in `api.ts`, so when you try to restore more than one file one after the other, it fails to get the latest commit after the first restore.

Reply:
You're right, but I think it doesn't immediately affect this feature since each commit only affects one file… which is then gone and doesn't need to be restored again.

Reply:
This still remains a problem... is there a way to specify when calling the function, that we don't want a cached result from `mem`?

Reply:
The closest solution I can think of is to use a cache buster 😬 
```js
	const {repository} = await api.v4(`
 		repository() { # Cache buster ${Math.random()}
 			pullRequest(number: ${getConversationNumber()!}) {
 				headRefOid
 			}
 		}
 	`);
```

A full fix might be too involved (e.g. replacing `mem` with a custom deduplicator in order to allow skipping the cache). This might be useful at some point, but let's not do it now.

Reply:
This cache busting might also be a solution to https://github.com/sindresorhus/refined-github/issues/4679 in `getBaseReference` (but can be handled in a separate PR)

Reply:
Added and it works!